libdnf/dnf-sack.cpp

```diff
@@ -2204,8 +2204,14 @@ static void setModuleExcludes(DnfSack *sack, std::vector<std::string> &includeNE
     transform(includeNEVRAs.begin(), includeNEVRAs.end(), includeNEVRAsCString.begin(), std::mem_fn(&std::string::c_str));
 
     libdnf::Query keepPackages{sack};
-    const char *keepRepo[] = {HY_CMDLINE_REPO_NAME, HY_SYSTEM_REPO_NAME, nullptr};
-    keepPackages.addFilter(HY_PKG_REPONAME, HY_NEQ, keepRepo);
+    std::vector<const char *> keepRepo;
+    keepRepo.push_back(HY_CMDLINE_REPO_NAME);
+    keepRepo.push_back(HY_SYSTEM_REPO_NAME);
+    for (auto repoid : hotfixRepos) {
```

Comment:
`for (auto & repoid : hotfixRepos) {`
repoid must be a reference to original value and not temporary copy. We need longer lifetime of value because we are storing result of c_str() (pointer into the repoid).

Reply:
fixed